api/src/router.rs

```diff
@@ -1,41 +1,43 @@
+use crate::endpoints::{book, identity, root};
 use crate::middleware;
+
 use warp::{filters::BoxedFilter, Filter, Reply};
 
 pub fn router() -> BoxedFilter<(impl Reply,)> {
-    warp::path::end()
+    let root = warp::path::end()
+        // GET - /
         .and(warp::get())
-        .and_then(crate::endpoints::root::root)
-        .boxed()
-        .or(warp::path("identity").and(
-            warp::path("oauth")
-                .and(
+        .and_then(root::root);
+    let identity = warp::path("identity").and(
+        warp::path("oauth")
+            .and(
+                warp::path("client_identifier")
+                    .and(warp::path::end())
                     // GET - /identity/oauth/client_identifier
-                    warp::path("client_identifier")
+                    .and(warp::get())
+                    .and_then(identity::get_oauth_client_identifier)
+                    .or(warp::path("authentication")
                         .and(warp::path::end())
-                        .and(warp::get())
-                        .and_then(crate::endpoints::identity::get_oauth_client_identifier)
-                        .boxed()
                         // POST - /identity/oauth/authentication
-                        .or(warp::path("authentication")
-                            .and(warp::path::end())
-                            .and(warp::post())
-                            .and(warp::body::json())
-                            .and_then(crate::endpoints::identity::create_oauth_authentication)
-                            .boxed()),
-                )
-                .boxed()
-                .or(warp::path("session")
-                    .and(
-                        // GET - /identity/session/info
-                        warp::path("info")
-                            .and(warp::path::end())
-                            .and(warp::get())
-                            .and(middleware::session::authorization())
-                            .and_then(crate::endpoints::identity::get_session_info)
-                            .boxed(),
-                    )
-                    .boxed()),
-        ))
+                        .and(warp::post())
+                        .and(warp::body::json())
+                        .and_then(identity::create_oauth_authentication)),
+            )
+            .or(warp::path!("session" / "info")
```

Comment:
The `path!()` macro will slow down compilation.

Reply:
... but makes the code cleaner.

And I actually didn't know that, but makes sense, since it first needs to be expanded.
I will make some time comparison.